src/components/Constants/UseApiConstants.tsx

```diff
@@ -7,9 +9,22 @@ import {
 export const useApiConstants = ():
   | LoadConstantsQuery['constant']
   | undefined => {
-  const { data } = useLoadConstantsQuery({
+  const { data, refetch } = useLoadConstantsQuery({
     fetchPolicy: 'cache-first',
   });
 
+  const userSavedLanguage = data?.user.preferences?.locale;
+  const [localeStorageLanguage, setLocaleStorageLanguage] = useLocalStorage(
+    `user-language`,
```

Comment:
What do you think about this to make it a little more clear that this value is supposed to be the language of the cached constants?

```suggestion
    'constants-language',
```